receiver/sqlserverreceiver/scraper.go

```diff
@@ -976,250 +945,92 @@ func (s *sqlServerScraperHelper) recordDatabaseSampleQuery(ctx context.Context)
+		openTransactionCountVal := s.retrieveValue(row, openTransactionCount, &errs, retrieveInt).(int64)
+		percentCompleteVal := s.retrieveValue(row, percentComplete, &errs, retrieveFloat).(float64)
+		queryStartVal := row[queryStart]
+		readsVal := s.retrieveValue(row, reads, &errs, retrieveInt).(int64)
+		requestStatusVal := row[requestStatus]
+		rowCountVal := s.retrieveValue(row, rowCount, &errs, retrieveInt).(int64)
+		sessionIDVal := s.retrieveValue(row, sessionID, &errs, retrieveInt).(int64)
+		sessionStatusVal := row[sessionStatus]
+		totalElapsedTimeMillisecondVal := s.retrieveValue(row, totalElapsedTimeMillisecond, &errs, retrieveIntAndConvert(func(i int64) any {
+			return float64(i) / 1000.0
+		})).(float64)
+		transactionIDVal := s.retrieveValue(row, transactionID, &errs, retrieveInt).(int64)
+		transactionIsolationLevelVal := s.retrieveValue(row, transactionIsolationLevel, &errs, retrieveInt).(int64)
+		usernameVal := row[username]
+		waitResourceVal := row[waitResource]
+		waitTimeMillisecondVal := s.retrieveValue(row, waitTimeMillisecond, &errs, retrieveIntAndConvert(func(i int64) any {
+			return float64(i) / 1000.0
+		})).(float64)
+		waitTypeVal := row[waitType]
+		writesVal := s.retrieveValue(row, writes, &errs, retrieveInt).(int64)
+
+		contextFromQuery := propagator.Extract(context.Background(), propagation.MapCarrier{
 			"traceparent": row[contextInfo],
-		}))
+		})
 
-		if spanContext.IsValid() {
-			record.SetTraceID(pcommon.TraceID(spanContext.TraceID()))
-			record.SetSpanID(pcommon.SpanID(spanContext.SpanID()))
-		} else {
-			attributes = append(attributes, internalAttribute{
-				key:            dbPrefix + contextInfo,
-				valueRetriever: defaultValueRetriever(hex.EncodeToString([]byte(row[contextInfo]))),
-				valueSetter:    setString,
-			})
-		}
+		spanContext := trace.SpanContextFromContext(contextFromQuery)
+		contextInfoVal := ""
 
-		for _, attr := range attributes {
-			value, err := attr.valueRetriever(row, attr.columnName)
-			if err != nil {
-				errs = append(errs, err)
-				s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing %s. original value: %s, err: %s", attr.columnName, row[attr.columnName], err))
-			}
-			attr.valueSetter(record.Attributes(), attr.key, value)
+		if !spanContext.IsValid() {
+			contextInfoVal = hex.EncodeToString([]byte(row[contextInfo]))
 		}
 
 		// client.address: use host_name if it has value, if not, use client_net_address.
 		// this value may not be accurate if
 		// - there is proxy in the middle of sql client and sql server. Or
 		// - host_name value is empty or not accurate.
+		var clientAddressVal string
 		if row[hostName] != "" {
-			record.Attributes().PutStr("client.address", row[hostName])
+			clientAddressVal = row[hostName]
 		} else {
-			record.Attributes().PutStr("client.address", row[clientAddress])
+			clientAddressVal = row[clientAddress]
 		}
 
-		s.lb.AppendLogRecord(record)
+		s.lb.RecordDbServerQuerySampleEvent(
```

Comment:
From what I can tell, all of the relevant `retrieveX` anonymous functions are properly converting units to the right "destination" unit (i.e. for attributes that are supposed to be in a unit of seconds, they're converted properly.)

However, the variable names here are mismatched to the actual unit they hold. They match the unit stored in SQL Server, but not what their value is. I think all of these names should be updated to match the proper attribute unit to avoid confusion.

Reply:
sorry my understanding could be wrong, but my feeling is most of the variables here do not necessarility have one `unit` in the exported attributes. so i am a little bit confused here, what `unit` should be used? take `logicalReads` or `sessionId` as examples, what unit should i use for those attributes? thanks

Reply:
thanks for the explanation. have updated it, please take another look. thanks!

Reply:
Yes, that's perfect, thanks for updating!